src/mixins/uploadFile.ts

```diff
@@ -0,0 +1,38 @@
+export interface UploadFileOptions {
+  multiple?: boolean;
+  accept?: string;
+}
+
+/**
+ * Mixin to add uploadFile method to component
+ */
+export const uploadFile = {
+  methods: {
+    /**
+     * Creates file input and requests from user a file to upload
+     *
+     * @param {UploadFileOptions} options
+     */
+    async uploadFile(options: UploadFileOptions = {}): Promise<FileList | null> {
+      const input = document.createElement('input');
+
+      input.type = 'file';
+
+      if (options.multiple) {
+        input.setAttribute('multiple', 'multiple');
+      }
+
+      if (options.accept) {
+        input.setAttribute('accept', options.accept);
+      }
+
+      return new Promise(resolve => {
+        input.addEventListener('change', () => {
+          resolve(input.files);
+        });
+
+        input.click();
```

Comment:
I remember, that in some `safari` versions, that won't work without adding input to DOM. 

https://github.com/codex-team/ajax/pull/15

You can remove element after `onchange` fired: https://github.com/codex-team/ajax/blob/master/src/utils.js#L119